packages/cli/docs/custom-plugins.md

```diff
@@ -0,0 +1,637 @@
+# Integrating a plugin in the CLI (based on real examples)
+
+One of the main features of Code PushUp is the ability to write custom plugins to track your own metrics.
+It enables you to implement nearly any kind of metric you want to track with minimum effort.
+In this section we will go through the steps to create a custom plugin and integrate it in your project.
+
+## Usage of a plugin in the core config
+
+To start crafting custom plugins you need a minimum `code-pushup.config.(ts|js|mjs)` file maintaining a `plugins` property.
+property.
+
+Any plugin is registered in a core config object and can take potential options to configure its behaviour.
+The following example shows where to register the plugin
+
+```typescript
+// code-pushup.config.ts
+import { create } from 'my-plugin';
+
+export default {
+  plugins: [
+    // can return `PluginConfig` or `Promise<PluginConfig>`
+    await create({
+      // plugin options here
+    }),
+  ],
+  categories: [],
+};
+```
+
+## Plugin Structure
+
+Every plugin is defined over a [`PluginConfig`](@TODO).
+
+The plugin config maintains:
+
+- metadata about the plugin
+- metadata about the available audits [`Audit`](@TODO)
+- internal logic producing the plugin output as [`AuditOutputs`](@TODO).
+
+A minimal plugin object maintaining the required fields looks like the following:
+
+**template for a custom plugin**
+
+```typescript
+// my-plugin.ts
+import { AuditOutputs, PluginConfig } from '@code-pushup/models';
+
+export const pluginMeta = {
+  slug: 'my-plugin',
+  title: 'My plugin',
+  // icon name from [vscode-material-icon-theme](https://github.com/PKief/vscode-material-icon-theme/tree/main/icons)
+  icon: 'javascript',
+};
+
+const auditMeta = {
+  slug: 'my-audit',
+  title: 'My audit',
+};
+
+export const audits = [auditMeta];
+
+export type Options = {
+  // your options here
+};
+
+export async function create(options: Options): PluginConfig {
+  return {
+    ...pluginMeta,
+    audits: [audit],
```

Comment:
On line 60, you define an `audits` variable. Shouldn't the value be directly added here? I don't see the variable being used anywhere.
```suggestion
    audits: [auditMeta],
```